packages/nest/src/require-flags-enabled.decorator.ts

```
@@ -0,0 +1,80 @@
+import type { CallHandler, ExecutionContext, HttpException, NestInterceptor } from '@nestjs/common';
+import { applyDecorators, mixin, NotFoundException, UseInterceptors } from '@nestjs/common';
+import type { Client } from '@openfeature/server-sdk';
+import { OpenFeature } from '@openfeature/server-sdk';
+
+/**
+ * Options for injecting a feature flag into a route handler.
+ */
+interface RequireFlagsEnabledProps {
+  /**
+   * The key of the feature flag.
+   * @see {@link Client#getBooleanValue}
+   */
+  flagKeys: string[];
+  /**
+   * The exception to throw if any of the required feature flags are not enabled.
+   * Defaults to a 404 Not Found exception.
```

I am not entirely sure why you chose 404 here. 
Is it to prevent unwanted information disclosure in the case the feature is meant to be hidden?

---

That's right. Defaulting to 404 as the preferred behaviour is to "hide" the existence of the Controller or Route from the caller entirely. This could also be integrated with any API documentation that the end user may generate.

---

Ya I actually think for security reasons this is the most reasonable default.